src/txmempool.cpp

```diff
@@ -441,12 +441,13 @@ bool CTxMemPool::addUnchecked(const uint256& hash, const CTxMemPoolEntry &entry,
     vTxHashes.emplace_back(hash, newit);
     newit->vTxHashesIdx = vTxHashes.size() - 1;
 
+    // Invalid ProTxes should never get this far because transactions should be
+    // fully checked by AcceptToMemoryPool() at this point, so we just assume that
+    // everything is fine here.
     if (tx.nType == TRANSACTION_PROVIDER_REGISTER) {
         CProRegTx proTx;
-        if (!GetTxPayload(tx, proTx)) {
-            LogPrintf("%s: ERROR: Invalid transaction payload, tx: %s", __func__, tx.ToString());
-            return false;
-        }
+        bool ok = GetTxPayload(tx, proTx);
```

Comment:
I would prefer `bool valid`

Reply:
IMO `valid` would be confusing because `GetTxPayload` doesn't validate anything, it just trying to deserialize things. So `success` is the proper one, but `ok` is shorter :)

Reply:
Yepp, agree. `true` could for example also be returned when a version (or any other integer) field is horribly out of range.